modules/claims_api/app/controllers/concerns/claims_api/disability_compensation_validations.rb

```diff
@@ -411,15 +414,28 @@ def validate_form_526_disability_secondary_disabilities!
     end
 
     def validate_form_526_disability_classification_code!
-      return if (form_attributes['disabilities'].pluck('classificationCode') - [nil]).blank?
+      form_attributes['disabilities'].each_with_index do |disability, index|
+        classification_code = disability['classificationCode']
+        next if classification_code.nil? || classification_code.blank?
+
+        if bgs_classification_ids.include?(classification_code)
+          validate_form_526_disability_classification_code_end_date!(classification_code, index)
+        else
+          raise ::Common::Exceptions::InvalidFieldValue.new("disabilities.#{index}.classificationCode",
+                                                            classification_code)
+        end
+      end
+    end
 
-      form_attributes['disabilities'].each do |disability|
-        next if disability['classificationCode'].blank?
-        next if bgs_classification_ids.include?(disability['classificationCode'])
+    def validate_form_526_disability_classification_code_end_date!(classification_code, index)
+      brd_disability = brd_disabilities.find { |d| d[:id] == classification_code.to_i }
+      end_date_time = brd_disability[:endDateTime] if brd_disability
```

Comment:
another fun nuance to us thinking we should have been / were calling BRD - we need to confirm the end date against the BGS call above, no reason to call BGS and then BRD. we'll make the entire shift when we move to BRD. BGS has the following as an example: 
```
{:name=>"ContentionClassification",
  :clsfcn_id=>"50",
  :clsfcn_txt=>"abscess, kidney",
  :end_dt=>"2016-02-04T11:51:56-06:00"},
```

I know the ticket said confirm against BRD /disabilities end date, but guess we need to resolve that.

Reply:
https://github.com/department-of-veterans-affairs/vets-api/pull/19868/commits/32286010ea22f0e7bb9eb9e2b774fdfad7c2008d